addon-resizer/nanny/main/pod_nanny.go

```diff
@@ -233,3 +237,99 @@ func decodeNannyConfiguration(data []byte, codecs *serializer.CodecFactory) (*na
 	}
 	return externalHC, nil
 }
+
+func updateResources(nannycfg *nannyconfig.NannyConfiguration, baseStorage string) []nanny.Resource {
+	var resources []nanny.Resource
+
+	// Monitor only the resources specified.
+	if nannycfg.BaseCPU != nannyconfig.NoValue {
+		resources = append(resources, nanny.Resource{
+			Base:         resource.MustParse(nannycfg.BaseCPU),
+			ExtraPerNode: resource.MustParse(nannycfg.CPUPerNode),
+			Name:         "cpu",
+		})
+	}
+
+	if nannycfg.BaseMemory != nannyconfig.NoValue {
+		resources = append(resources, nanny.Resource{
+			Base:         resource.MustParse(nannycfg.BaseMemory),
+			ExtraPerNode: resource.MustParse(nannycfg.MemoryPerNode),
+			Name:         "memory",
+		})
+	}
+
+	if baseStorage != nannyconfig.NoValue {
+		resources = append(resources, nanny.Resource{
+			Base:         resource.MustParse(baseStorage),
+			ExtraPerNode: resource.MustParse(nannycfg.MemoryPerNode),
+			Name:         "storage",
+		})
+	}
+
+	glog.Infof("Resources: %+v", resources)
+	return resources
+}
+
+type nannyConfigUpdater struct {
+	clientset          *kubernetes.Clientset
+	defaultConfig      *nannyconfigalpha.NannyConfiguration
+	namespace          string
+	nannyConfigMapName string
+	baseStorage        string
+}
+
+// newNannyConfigUpdater gives a NannyConfigUpdater
+// with the given dependencies.
+func newNannyConfigUpdater(clientset *kubernetes.Clientset, defaultConfig *nannyconfigalpha.NannyConfiguration, namespace, nannyConfigMapName, baseStorage string) nanny.NannyConfigUpdater {
+	result := &nannyConfigUpdater{
+		clientset:          clientset,
+		defaultConfig:      defaultConfig,
+		namespace:          namespace,
+		nannyConfigMapName: nannyConfigMapName,
+		baseStorage:        baseStorage,
+	}
+	return result
+}
+
+// CurrentResources fetches latest data from NannyConfiguration
+// through API calls and returns required resources.
+func (n *nannyConfigUpdater) CurrentResources() ([]nanny.Resource, error) {
+	_, codecs, err := nannyscheme.NewSchemeAndCodecs()
+	if err != nil {
+		return nil, err
+	}
+	// overwrite defaults with flag-specified parameters
+	nannyconfigalpha.SetDefaults_NannyConfiguration(n.defaultConfig)
+	// retrieve config map parameters if present
+	configMapConfig := &nannyconfigalpha.NannyConfiguration{}
+
+	nannycfg, err := n.clientset.CoreV1().ConfigMaps(n.namespace).Get(context.Background(), n.nannyConfigMapName, metav1.GetOptions{})
```

Comment:
Good catch! Yes it will return Not Found error I believe. I think in this case we shouldn't return error and quit addon resizer. We should use default parameters, similar to what happens when failing to read data from file path in original loadNannyConfiguration function.

Updated the code to log the error and use default parameter.